scripts/create-mermaid-diagram.sh

```diff
@@ -0,0 +1,56 @@
+# ensure yj is present
+if [ ! -f ~/yj ]; then
```

Comment:
why `~/yj` ? shouldn't you check if in path first otherwise should be something like `/tmp/yj`

Reply:
Yeah, thought about that re. getting it working on Mac. 

Re. using ~/yj. It's because of an issue getting it installed on the GitHub Actions runner. I tried putting it in /usr/bin/ and /bin and chmod-ing to make it executable but it just wouldn't work. Brew won't install it in the same place across mac and linux so I figured to just get the binary and place it in a consistent directory. Could look to see if there is an action for installing via brew maybe.

Reply:
There is a github action that allows you to install brew so the script now works with yj being in a non-hardcoded path. Works now on macOS and Linux (albeit requiring brew to be installed)

Reply:
I would just stick it in /tmp or install it in the github action and require that people running it manually install it first however they want to.

magic bites people, and 'littering' is annoying